messaging/src/main/java/io/lantern/messaging/Messaging.kt

```diff
@@ -353,6 +353,39 @@ class Messaging(
         }
     }
 
+    /**
+     * Re-sends the failed message identified by messageId to all recipients to whom we were unable
+     * to send the message.
+     *
+     * throws IllegalArgumentException if the message couldn't be found, wasn't sent by us, is
+     *        already in the process of sending or was already successfully sent.
+     */
+    @Throws(IllegalArgumentException::class)
+    fun resendFailedMessage(messageId: String) {
+        db.mutate { tx ->
+            tx.get<Model.StoredMessage>(myId.id.storedMessagePath(messageId)).let { msg ->
+                if (msg == null) {
```

Comment:
I think that this validation for null checking is redundant, since we are using Kotlin for the Messaging library, we can reduce the validation to something like this.

```kotlin
tx.get<Model.StoredMessage>(myId.id.storedMessagePath(messageId))?.run {
                when (status) {
                    Model.StoredMessage.DeliveryStatus.SENDING ->
                        throw IllegalArgumentException("Message is still in the process of sending")
                    Model.StoredMessage.DeliveryStatus.PARTIALLY_SENT ->
                        throw IllegalArgumentException("Message is still in the process of sending")
                    Model.StoredMessage.DeliveryStatus.COMPLETELY_SENT ->
                        throw IllegalArgumentException("Message was already successfully sent")
                }

                val out = Model.OutboundMessage.newBuilder().setMessageId(messageId)
                    .setSent(now)
                    .setSenderId(myId.id)
                    .setRecipientId(contactId.id)
                tx.put(out.dbPath, out.build())
                cryptoWorker.submit { cryptoWorker.processOutbound(out) }
            } ?: run {
                    throw java.lang.IllegalArgumentException("Message not found")
            }
```

instead of using the let and assign a variable to handle all the scope, we can use run to access directly the content, and since we are validating that the data isn't null we can safety retrieve any information that we need without having to create a holder variable and then check if that holder is null or not. And finally for the previous if(msg == null) at the end we just add ?: run which only runs when the previous information is null.

Reply:
Ha, that run syntax is pretty neat, I didn't know you could do that. I think I still prefer let, because I like being clear about the origin of those variables.

BTW, I originally used `?.let {} ?: run` but decided to do it this way so that all the validation happens in one place.

Reply:
It's a thing of preferences, in my case I prefer using this ```?.let {} ?: run``` if I have to do alterations, but if im only using internal calls then the ```run``` is good to go, leaving that aside, this code is ready to be served. @oxtoacart

Reply:
Yeah, I can see how it's nice. I'm already using `let` all over the place so I'm going to stick with that style for now.